src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/LintTask.kt

```diff
@@ -26,66 +25,38 @@ import org.jmailen.gradle.kotlinter.tasks.lint.LintWorkerRunnable
 @CacheableTask
 open class LintTask @Inject constructor(
     private val workerExecutor: WorkerExecutor
-) : SourceTask() {
+) : ConfigurableKtLintTask() {
 
+    @Optional
     @OutputFiles
-    lateinit var reports: Map<String, File>
+    val reports = mapProperty(default = emptyMap<String, File>())
```

Comment:
Changed behavior: 
Previously, uninitialized reports throwed exception. Now the task simply skips report generation.

Reply:
Specifying at least one report output should be mandatory. It uses the report outputs to know if the tasks are up-to-date. Also, no reporting artifacts is probably never what the user wants and they likely got there only by mistake.

.editorconfig of course can remain optional.

Reply:
>Specifying at least one report output should be mandatory

Thanks for the explanation! So what should happen if user passes an empty map `reports = [:]`?
If I remember well I did such thing in my project because I didn't care about any file outputs. I just wanted the build to fail. 

> It uses the report outputs to know if the tasks are up-to-date

Can you point out the place where those reports are marked as task input?
Plus, as far as I can see LintTask passes its own `HasErrorReporter` so there always would be at least one reporter.

Reply:
`[:]` would also be an error I think. In your project, is your `lintKotlin` task ever "UP-TO-DATE"?

`reports` is annotated as `@OutputFiles` which means in factors in to incremental build. You should see this in a normal project with at least one report.
1. `gradle -i lintKotlin` from a clean state and notice it runs the task
2. `gradle -i lintKotlin` again and notice the task is up to date and does not rerun
3. delete the report output files and repeat the process noting the task reruns

https://docs.gradle.org/current/userguide/more_about_tasks.html

Reply:
> In your project, is your lintKotlin task ever "UP-TO-DATE"?

I confirm all tasks normally became `UP-TO-DATE` 🤷‍♂ I've also commited new tests confirming that behavior. 

> You should see this in a normal project with at least one report.

I confirm that's what happens with an empty reports map. (Except the `delete the report output files` part as I cannot delet any files because they don't exist ofc 😛)

I think you're referring to the behavior change introduced in: https://github.com/gradle/gradle/pull/9896 but isn't applicable here. The `LintTask`, both in previous version as the one I'm suggesting declares Outputs and in both cases they both properly handles empty map case (so the output is declared, it exists, but it is empty so it never changes) therefore Gradle takes only task inputs into account when running incremental build. 
Documentation seems to confirm that behavior:
> As part of incremental build, Gradle tests whether any of the task inputs or outputs **has** **changed** since the last build. If they haven’t, Gradle can consider the task up to date and therefore skip executing its actions. Also note that incremental build won’t work unless a task has **at least one task output**, although tasks usually have at least one input as well.

and I would name that consistent with the fact that `ListProperty` and `MapProperty` always have default values where other properties throw exception when accessed uninitialized (unless marked as `@Optional`.), having an empty collection is considered a valid case.
Or at least that's my understanding of how `Inputs` and `Outputs` work 🤷‍♂ 

Therefore, if the reports property was required only for the sake of incremental build, and empty map has been treated as a valid value I'd like to propose to keep that behavior and unify it with the _uninitialized_ value case. That would be somehow backward compatbile, right? Every developer that had an empty map passed would still have their code working. There are no developers who have reports set to `null` as that previously resulted in an exception.

Reply:
If an empty value doesn't interfere with incremental build and some developers find it preferable, I'm good with keeping it.

Generally I think output artifacts are a good idea for almost any task -- a build step produces _something_ after all. Many people are using these reports to display the build results in CI.